backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -531,9 +600,9 @@ func initLogArchive() (logArchive archive.LogArchiveInterface) {
 }
 
 // NewClientManager creates and Init a new instance of ClientManager.
-func NewClientManager() ClientManager {
+func NewClientManager(useKubernetesStorage bool) ClientManager {
```

Comment:
Since this is technically a breaking API change since this is a public method, could make this accept a struct called `options` or something so that we can add more options in the future. Also, this boolean should likely mention `pipeline` in the name.